.github/workflows/check_files.py

```diff
@@ -0,0 +1,159 @@
+    base_branch = f"origin/{base_branch}"
+    pr_branch = f"origin/{pr_branch}"
+
+    command = f"git diff --name-only {base_branch}...{pr_branch} --diff-filter=ACMRT | xargs"
+
+    try:
+        # Run git command to get the list of changed files
+        process = subprocess.Popen(
+            command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
+        )
+        output, error = process.communicate()
+    except Exception as e:
+        print(f"Error: {e}")
+        sys.exit(1)
+
+    changed_files = [file.strip() for file in output.split(' ') if file.strip()]
+    file_count = len(changed_files)
+    return ScriptResult(file_count=file_count, unauthorized_changes=changed_files)
+
+def _check_unauthorized_changes(changed_files, sensitive_files):
+    """
+    Check for unauthorized file changes.
+    Args:
+        changed_files (list): List of changed files.
+        sensitive_files (list): List of sensitive files.
+    Returns:
+        list: Unauthorized changes.
+
+    """
+
+    unauthorized_changes = [file for file in changed_files if any(glob.fnmatch.fnmatch(file, sf) for sf in sensitive_files if sf)]
+    return ScriptResult(len(unauthorized_changes), unauthorized_changes)
+    
+
+def _arg_parser_resolver():
+    """Resolve the CLI arguments provided by the user.
+    Args:
+        None
+    Returns:
+        result: Parsed argument object
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--base_branch",
+        type=str,
+        required=True,
+        help="Base branch where a pull request should be made."
+    )
+    parser.add_argument(
+        "--pr_branch",
+        type=str,
+        required=True,
+        help="PR branch from where the pull request is made.",
+    )
+    parser.add_argument(
+        "--file_count",
+        type=int,
+        default=20,
+        help="Number of files changes allowed in a single commit"
+    )
+    parser.add_argument(
+        "--sensitive_files",
+        type=str,
+        required=True,
+        help="Path to a file containing a list of sensitive files and directories."
+    )
+    return parser.parse_args()
+
+def main():
+    """
+    Execute the script's main functionality.
+    This function serves as the entry point for the script. It performs
+    the following tasks:
+    1. Validates and retrieves the base branch and PR commit from
+       command line arguments.
+    2. Counts the number of changed files between the specified branches.
+    3. Checks if the count of changed files exceeds the acceptable
+       limit (20).
+    4. Checks for unauthorized file changes based on a list of sensitive files.
+    5. Provides informative messages based on the analysis.
+    Raises:
+        SystemExit: If an error occurs during execution.
+    """
+
+    args = _arg_parser_resolver()
+
+    base_branch = args.base_branch
+    pr_branch = args.pr_branch
+
+    print(f"You are trying to merge on branch: {base_branch}")
```

Comment:
Only print non error messages if an argparse `--verbose`flag is set. Do this for all cases